dipy/reconst/shm.py

```diff
@@ -230,164 +232,205 @@ def real_sph_harm(m, n, theta, phi):
     return real_sh
 
 
-def real_sym_sh_mrtrix(sh_order, theta, phi):
-    """
-    Compute real symmetric spherical harmonics as in Tournier 2007 [2]_, where
-    the real harmonic $Y^m_n$ is defined to be::
+def real_sh_tournier(sh_order, theta, phi,
+                     is_full_basis=False,
+                     is_legacy=False):
+    """ Compute real spherical harmonics as initially defined in Tournier
+    2007 [1]_ then updated in MRtrix [2]_, where the real harmonic $Y^m_n$
+    is defined to be:
 
-        Real($Y^m_n$)       if m > 0
-        $Y^0_n$             if m = 0
-        Imag($Y^|m|_n$)     if m < 0
+        Real($Y^m_n$) * sqrt(2)      if m > 0
+        $Y^0_n$                      if m = 0
+        Imag($Y^|m|_n$) * sqrt(2)    if m < 0
 
     This may take scalar or array arguments. The inputs will be broadcasted
     against each other.
 
     Parameters
-    -----------
+    ----------
     sh_order : int
         The maximum degree or the spherical harmonic basis.
     theta : float [0, pi]
         The polar (colatitudinal) coordinate.
     phi : float [0, 2*pi]
         The azimuthal (longitudinal) coordinate.
+    is_full_basis: bool, optional
+        If true, returns a basis including odd order SH functions as well as
+        even order SH functions. Otherwise returns only even order SH
+        functions. Default: False
+    is_legacy: bool, optional
```

Comment:
I prefer to be explicit here. How about calling this keyword argument `mrtrix3`, set to `True` per default?

Reply:
I agree it would be less confusing, but it wouldn't work with the current dictionary called in `sh_to_sf` and `sh_to_sf_matrix` because the meaning of the legacy and mrtrix3 arguments are different. Maybe this could justify replacing the dictionary by a method as suggested by @jhlegarreta [here](https://github.com/dipy/dipy/pull/2206#issuecomment-691522783). I'm also starting to doubt the pertinence of the `legacy` argument itself. Maybe the legacy implementations should be methods of their own, with a unique basis name, that we could remove in the future when they are deprecated. This would involve less maintenance as we won't have to change `real_sh_tournier` and `real_sh_descoteaux` when the legacy bases get deprecated. Opinions on this?